src/goto-instrument/reachability_slicer.cpp

```diff
@@ -55,6 +62,45 @@ void reachability_slicert::fixedpoint_assertions(
   }
 }
 
+/// Perform forwards breadth-first-search of the control-flow graph of the
```

Comment:
If that's what's happening, why not use `grapht` functionality to accomplish it?

Reply:
No idea. I was just modifying the code that was there already. I'd like to avoid making bigger changes, I don't know the codebase well enough to guarantee that I'm not doing more harm than good.

Reply:
I'd suggest the following:
1. Change `grapht::get_reachable` to take a `std::vector` or a `std::set` of `node_indext` as the first parameter.
2. Implement `fixedpoint_from_assertions` as follows:
````c++
for(const auto &cfg_entry : cfg.entry_map)
{
  std::vector<cfgt::node_indext> queue;
  if(criterion(cfg_entry.first) || is_threaded(cfg_entry.first))
    queue.push_back(cfg_entry.second);

  std::vector<cfgt::node_indext> reachable = cfg.get_reachable(queue, true);
  for(const auto index : reachable)
    cfg[index].reachable_from_assertion = true;
}
````

And then `reachability_slicert::fixedpoint_to_assertions` could also be rewritten in the same way, just setting the second argument of `get_reachable` to `false`.